Sources/Sentry/SentryClient.m

```diff
@@ -887,6 +871,25 @@ - (SentryEvent *_Nullable)prepareEvent:(SentryEvent *)event
         }
     }
 
+    if (event != nil) {
```

Comment:
`m`: I think it would be good to add a short comment explaining why we need to call this code at this location.